sympy/solvers/solveset.py

```diff
@@ -978,6 +985,318 @@ def _solveset(f, symbol, domain, _check=False):
+    r"""
+    Helper to check whether an equation is exponential or not.
+
+    ``f`` is the equation to be checked
+
+    ``symbol`` is the variable in which the equation needs to be checked
+
+    The function extracts each term of the equation and checks if it is
+    of exponential form w.r.t `symbol`.
+    Returns `True` if any term is of exponential type otherwise `False`.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, cos, exp
+    >>> from sympy.solvers.solveset import _check_expo as check
+    >>> x, y = symbols('x y')
+    >>> check(x**y - x, y)
+    True
+    >>> check(x**y - x, x)
+    False
+    >>> check(exp(x + 3), x)
+    True
+    >>> check(cos(2**x), x)
+    False
+    """
+
+    for add_arg in Add.make_args(f):
+        for mul_arg in Mul.make_args(add_arg):
+            if isinstance(mul_arg, (Pow, exp)) and (
+                    symbol in mul_arg.exp.free_symbols):
+                return True
+    return False
+
+
+def _transolve(f, symbol, domain):
+    r"""
+    Function to solve transcendental equations. It is a helper to
+    `solveset` and should be used internally. `\_transolve`
+    currently supports the following class of equations:
+
+        - Exponential equations
+
+    Parameters
+    ==========
+
+    f : is any transcendental equation that needs to be solved.
+        This needs to be an expression, which is assumed
+        to be equal to 0.
+
+    symbol : is variable for which the equation is solved.
+        This needs to be of class `Symbol`.
+
+    domain : is a set over which the equation is solved.
+        This needs to be of class `Set`.
+
+    How to use `\_transolve`
+    ========================
+
+    `\_transolve` should not be used as an independent function, because
+    it assumes that the equation (`f`) and the `symbol` comes from
+    `solveset` and might have undergone a few modification(s).
+    To use `\_transolve` as an independent function the equation (`f`)
+    and the `symbol` should be passed as they would have been by
+    `solveset`.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import _transolve as transolve
+    >>> from sympy.solvers.solvers import _tsolve as tsolve
+    >>> from sympy import symbols, S, pprint
+    >>> x = symbols('x', real=True) # assumption added
+    >>> transolve(5**(x - 3) - 3**(2*x + 1), x, S.Reals)
+    {-log(375)/(-log(5) + 2*log(3))}
+
+    How `\_transolve` works
+    =======================
+
+    `\_transolve` uses two types of helper functions to solve equations
+    of a particular class:
+
+    Identifying helpers: These helpers are used to determine whether a
+    given equation belongs to a certain class of equation or not.
+    These functions return either True or False.
+
+    Solving helpers: These are the helpers that solve the equation.
+    They are invoked once the equation is identified by the `identifying
+    helpers`. These functions return either the complete solution or
```

Comment:
If the solving helper only takes as input the function, how will it be able to return a solution? (c.f. line 1239 below)

Reply:
They won't be able to return a solution, I would rather remove "complete solution". The idea now is that the solving helpers will convert the equation into a tractable form and then in the `_transolve` itself, the changed equation can be handled. This is how `_solve_expo` works and we will make the other helpers work in similar fashion. For this scenario I guess having the `equation (f)` as the only argument for `solving helpers` would work?